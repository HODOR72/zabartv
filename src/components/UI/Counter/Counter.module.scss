.counter {
	display: flex;
	align-items: center;
}

.decrement,
.increment {
	display: block;
	width: 28px;
	height: 28px;
	background-color: #303847;
	backdrop-filter: blur(25px);
	border-radius: 100%;
	transition: background-color 0.3s, opacity 0.3s;

	&[disabled] {
		pointer-events: none;
		opacity: 0.5;
	}

	&:hover {
		background-color: #465165;
	}

	&:before,
	&:after {
		content: '';
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		position: absolute;
		border-radius: 50px;
		background-color: #fff;
	}

	&:before {
		width: 7px;
		height: 1.5px;
	}

	&:after {
		width: 1.5px;
		height: 7px;
	}
}

.decrement {
	&:after {
		display: none;
	}
}

.value {
	display: block;
	text-align: center;
	min-width: 100px;
	font-size: 20px;
	line-height: 36px;
	color: var(--color-white);
	transition: color 0.3s;
}
