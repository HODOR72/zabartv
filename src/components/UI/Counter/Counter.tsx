import { ButtonBase } from '@/components/ButtonBase/ButtonBase';
import classNames from 'classnames';
import { FC, useEffect, useState } from 'react';
import styles from './Counter.module.scss';

interface CounterProps {
	initialValue: number;
	caption?: string;
	className?: string;
}

export const Counter: FC<CounterProps> = ({ initialValue, caption, className }) => {
	const [value, setValue] = useState<number>(initialValue);

	useEffect(() => {
		localStorage.setItem('counter', String(value));
	}, [value]);

	const handleDecrement = () => setValue(value - initialValue);
	const handleIncrement = () => setValue(value + initialValue);

	return (
		<div data-testid="counter" className={classNames(styles.counter, className)}>
			<ButtonBase
				data-testid="counter-decrement"
				onClick={handleDecrement}
				className={styles.decrement}
				aria-label="Минус"
				disabled={value === initialValue}
			/>
			<span data-testid="counter-value" className={styles.value}>
				{value} {caption}
			</span>
			<ButtonBase
				data-testid="counter-increment"
				onClick={handleIncrement}
				className={styles.increment}
				aria-label="Плюс"
			/>
		</div>
	);
};
