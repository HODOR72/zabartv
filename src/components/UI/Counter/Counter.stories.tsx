import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Counter } from './Counter';

export default {
	title: 'Counter',
	component: Counter,
} as ComponentMeta<typeof Counter>;

const Template: ComponentStory<typeof Counter> = (args) => <Counter {...args} />;

export const Example = Template.bind({});
Example.args = {
	initialValue: 30,
	caption: 'Days',
};
