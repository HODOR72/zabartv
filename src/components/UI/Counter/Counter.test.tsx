import { render, screen, fireEvent } from '@testing-library/react';
import { Counter } from './Counter';

describe('Counter.test.tsx', () => {
	const Wrapper = () => {
		return <Counter initialValue={60} />;
	};

	test('increment', () => {
		render(<Wrapper />);
		const element = screen.getByTestId('counter-increment');
		const value = screen.getByTestId('counter-value');

		fireEvent.click(element);

		expect(value).toHaveTextContent('120');

		expect(screen.getByTestId('counter')).toMatchSnapshot();
	});

	test('decrement', () => {
		render(<Wrapper />);
		const element = screen.getByTestId('counter-decrement');
		const value = screen.getByTestId('counter-value');

		fireEvent.click(element);

		expect(value).toHaveTextContent('60');

		expect(screen.getByTestId('counter')).toMatchSnapshot();
	});
});
