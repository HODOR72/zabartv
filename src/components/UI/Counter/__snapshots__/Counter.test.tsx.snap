// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.test.tsx decrement 1`] = `
<div
  class="counter"
  data-testid="counter"
>
  <button
    aria-label="Минус"
    class="btn decrement"
    data-testid="counter-decrement"
    disabled=""
  />
  <span
    class="value"
    data-testid="counter-value"
  >
    60
     
  </span>
  <button
    aria-label="Плюс"
    class="btn increment"
    data-testid="counter-increment"
  />
</div>
`;

exports[`Counter.test.tsx increment 1`] = `
<div
  class="counter"
  data-testid="counter"
>
  <button
    aria-label="Минус"
    class="btn decrement"
    data-testid="counter-decrement"
  />
  <span
    class="value"
    data-testid="counter-value"
  >
    120
     
  </span>
  <button
    aria-label="Плюс"
    class="btn increment"
    data-testid="counter-increment"
  />
</div>
`;
